x-pack/apm-server/sampling/eventstorage/storage_manager.go

```diff
@@ -107,16 +115,121 @@ func (s *StorageManager) runValueLogGC(discardRatio float64) error {
 	return s.db.RunValueLogGC(discardRatio)
 }
 
+// RunDropLoop runs a loop that detects if storage limit has been exceeded for at least ttl.
+// If so, it drops and recreates the underlying badger DB.
+// The loop stops when it receives from stopping.
+func (s *StorageManager) RunDropLoop(stopping <-chan struct{}, ttl time.Duration, storageLimitInBytes uint64) error {
```

Comment:
I might be missing something but why do we need to expose the gc loop and this loop? Can we start these when a new storage manager is created and manage its lifecycle as per the lifecycle of the storage manager?

Reply:
This is for EA hot reload. I should add a comment to explain it. DropLoop and gc loop need to use the config supplied by EA, as this piece of info would not be available at `main.go` initialization time.

Reply:
Hmm, it still feels a bit excessive to expose these outside the storage manager. Will it be possible to have a method like `Run` which can be called multiple times and wrap both the loops inside that?

Reply:
Done, put both GCLoop and DropLoop inside Run.